contracts/SpokePool.sol

```diff
@@ -636,10 +634,39 @@ abstract contract SpokePool is
         uint32 fillDeadline,
         bytes memory message
     ) public payable override nonReentrant unpausedDeposits {
-        // Validate
+        // Check that deposit route is enabled for the input token. There are no checks required for the output token
+        // which is pulled from the relayer at fill time and passed through this contract atomically to the recipient.
+        require(enabledDepositRoutes[inputToken.token][destinationChainId], "Disabled route");
 
-        // Do stuff
-        // - Pull funds from depositor
+        //@dev The following checks are copied from _deposit. Are they still needed?
+        require(inputToken.amount <= MAX_TRANSFER_SIZE, "Amount too large");
+        require(outputToken.amount <= MAX_TRANSFER_SIZE, "Amount too large");
+
+        // Require that quoteTimestamp has a maximum age so that depositors pay an LP fee based on recent HubPool usage.
+        // It is assumed that cross-chain timestamps are normally loosely in-sync, but clock drift can occur. If the
+        // SpokePool time stalls or lags significantly, it is still possible to make deposits by setting quoteTimestamp
+        // within the configured buffer. The owner should pause deposits if this is undesirable. This will underflow if
+        // quoteTimestamp is more than depositQuoteTimeBuffer; this is safe but will throw an unintuitive error.
+
+        // slither-disable-next-line timestamp
+        require(getCurrentTime() - quoteTimestamp <= depositQuoteTimeBuffer, "invalid quoteTimestamp");
+
+        // fillDeadline is relative to the destination chain.
+        // Don’t allow fillDeadline to be more than ~3 bundles into the future, so 12 hours maybe.
+        // This way the dataworker/relayer doesn’t have to maintain more a lookback longer than this.
+        // TODO: We could choose to skip this check in favor of adding a rule in the dataworker to enforce this
+        // constraint.
+        require(fillDeadline <= getCurrentTime() + 12 hours, "invalid fillDeadline");
+
+        // If the address of the origin token is a wrappedNativeToken contract and there is a msg.value with the
+        // transaction then the user is sending ETH. In this case, the ETH should be deposited to wrappedNativeToken.
+        if (inputToken.token == address(wrappedNativeToken) && msg.value > 0) {
```

Comment:
Not sure how we didn't see this earlier, but wrappedNativeToken should also be immutable, since this basically never changes.

Reply:
So good point, like `depositQuoteTimeBuffer` above, but I think we have a problem converting this to an immutable variable. Immutable variables must be set in constructors or inline, which presents problems if we want a different value set here for each of the spokepools. Additionally there might be other issues with immutable variables in upgradeable contracts https://forum.openzeppelin.com/t/upgradable-contracts-instantiating-an-immutable-value/28763

Reply:
https://docs.openzeppelin.com/upgrades-plugins/1.x/faq#why-cant-i-use-immutable-variables

Reply:
I think OZ's docs are good for the common case, but they _don't apply to this use case_, assuming the tools still work with constructors and immutable variables in the implementation contract.

1. OZ says constructors should be replaced with initializers, which is true! However, I don't see why the implementation contract can't still have a constructor that sets immutable variables in the bytecode.
2. OZ says this means that the immutable variable will be shared by all proxies that point at the implementation. In this case, that's fine, desirable even! There is only have 1 proxy pointing at 1 implementation per chain and the implementation can be upgraded anytime to change the variable.

Reply:
As per our offline discussions, https://github.com/across-protocol/contracts-v2/commit/be1c03926b89280542a94b566166628a5000ae98 implements the gas optimizations that convert storage vars to immutable/constant ones.